examples/js/exporters/GLTFExporter.js

```diff
@@ -707,8 +709,13 @@ THREE.GLTFExporter.prototype = {
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
 
-				canvas.width = image.width;
-				canvas.height = image.height;
+				if ( image.width > options.maxTextureWidth || image.height > options.maxTextureHeight ) {
+
+					console.warn( 'GLTFExporter: Resized too big image.', image );
+				}
+
+				canvas.width = Math.min( image.width, options.maxTextureWidth );
+				canvas.height = Math.min( image.height, options.maxTextureHeight );
 
 				if ( options.forcePowerOfTwoTextures && ! isPowerOfTwo( image ) ) {
```

Comment:
I believe we should check `isPowerOfTwo( canvas )` in case the user has set a `maxTextureWidth/Height` that are not POT and we have resized the canvas to that new size.